instrumentation/cassandra/cassandra-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v3_0/TracingSession.java

```diff
@@ -48,97 +48,110 @@ public Session init() {
 
   @Override
   public ResultSet execute(String query) {
-    Context context = tracer().startSpan(Context.current(), session, query);
+    CassandraRequest request = new CassandraRequest(session, query);
+    Context context = instrumenter().start(Context.current(), request);
     ResultSet resultSet;
     try (Scope ignored = context.makeCurrent()) {
       resultSet = session.execute(query);
     } catch (Throwable t) {
-      tracer().endExceptionally(context, t);
+      instrumenter().end(context, request, null, t);
       throw t;
     }
-    tracer().end(context, resultSet.getExecutionInfo());
+    request.setExecutionInfo(resultSet.getExecutionInfo());
+    instrumenter().end(context, request, null, null);
```

Comment:
do you think this justifies adding `RESPONSE` type to `DbAttributesExtractor`?

Reply:
`executionInfo` seems to only be used in net attributes, so it can have a `RESPONSE` I think?

Reply:
While working on gRPC instrumentation I understand what you're saying now :) Yeah I think it should, we should just have RESPONSE on all extractors it seems

Reply:
> do you think this justifies adding `RESPONSE` type to `DbAttributesExtractor`?

👍 
I think that none of our "default" attribute extractors should impose any type restrictions on the user; we should definitely use `RESPONSE` instead of hard-coded `Void`.

Reply:
I added issue #3076 and will do this as separate PR, since it will conflict with a few PRs currently in flight.